daemon/attach.go

```diff
@@ -60,6 +60,7 @@ func (daemon *Daemon) ContainerAttach(prefixOrName string, c *backend.ContainerA
 
 	if err := daemon.containerAttach(container, stdin, stdout, stderr, c.Logs, c.Stream, keys); err != nil {
 		fmt.Fprintf(outStream, "Error attaching: %s\n", err)
+		return err
```

Comment:
@allencloud  If we want the client receive non-zero exit code once `daemon.containerAttach` return error, I think  passing `GetStreams` to `daemon.containerAttach` and  do `GetStreams` just before io copy may be a way, so any error happened before io copy can return to client. Once io copy start, the the io streams is from container.  Correct me if I'm wrong :)